assembly/src/lib.rs

```diff
@@ -163,18 +164,22 @@ impl AsRef<str> for AbsolutePath {
 }
 
 impl Serializable for AbsolutePath {
-    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
-        target.write_str(&self.path)
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.len() as u64);
+        target.write_bytes(self.as_bytes());
     }
```

Comment:
I think `u64` is too much - let's use `u16` and add TODOs to enforce this restriction via the constructor.